src/engine/input/key-mapping/InputGroupList.cs

```diff
@@ -0,0 +1,180 @@
+﻿using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using Godot;
+using Newtonsoft.Json;
+
+/// <summary>
+///   Used by OptionsMenu>Inputs>InputGroupContainer
+/// </summary>
+public class InputGroupList : VBoxContainer
+{
+    [Export]
+    public NodePath ConflictDialogPath;
+
+    internal static readonly PackedScene InputEventItemScene =
+        GD.Load<PackedScene>("res://src/engine/input/key-mapping/InputEventItem.tscn");
+
+    private static readonly Dictionary<string, List<InputEventWithModifiers>> DefaultControlScheme =
+        GetCurrentlyAppliedControlScheme();
+
+    private ConfirmationDialog conflictDialog;
+
+    private InputEventItem latestDialogCaller;
+    private InputEventItem latestDialogConflict;
+    private InputEventWithModifiers latestDialogNewEvent;
+
+    public InputGroupList()
+    {
+        Instance = this;
+    }
+
+    public delegate void ControlSchemeChangedDelegate(Dictionary<string, List<InputEventWithModifiers>> data);
+
+    public event ControlSchemeChangedDelegate OnControlSchemeChanged;
+
+    public static InputGroupList Instance { get; private set; }
+    public IEnumerable<InputGroupItem> AllGroupItems => GetChildren().OfType<InputGroupItem>();
```

Comment:
If children are needed to be accessed often they should be kept in a separate list when adding them.